src/plugins/next-env/plugin.ts

```diff
@@ -15,18 +15,23 @@ export function vitePluginNextEnv(
 
   const resolvedDir = resolve(rootDir);
 
-  let getDefineEnv: typeof import("next/dist/build/define-env.js").getDefineEnv;
-  let isNext1540 = false;
-
-  try {
-    // Next.js >= 15.4.0
-    getDefineEnv = require("next/dist/build/define-env.js").getDefineEnv;
-    isNext1540 = true;
-  } catch (error) {
-    // Next.js < 15.4.0
-    getDefineEnv =
-      require("next/dist/build/webpack/plugins/define-env-plugin.js").getDefineEnv;
-  }
+  // Try Next.js >= 15.4.0
+  const getDefineEnvModule = import("next/dist/build/define-env.js")
```

Comment:
@elecdeer 
Hi👋 I was also running into this same issue!
I think it would be better to use `require` created by `createRequire` to maintain consistency with the other code in the codebase.

https://github.com/storybookjs/vite-plugin-storybook-nextjs/blob/main/src/plugins/next-mocks/compatibility/utils.ts#L4